lib/src/rules/missing_whitespace_between_adjacent_strings.dart

```diff
@@ -95,44 +95,34 @@ class _Visitor extends RecursiveAstVisitor<void> {
 }
 
 extension on StringLiteral {
-  /// Returns whether this ends with whitespace, where any
+  /// Returns whether this ends with whitespace, where an initial
   /// [InterpolationExpression] counts as whitespace.
   bool get endsWithWhitespace {
-    if (this is SimpleStringLiteral) {
-      return (this as SimpleStringLiteral).value.endsWithWhitespace;
-    } else if (this is StringInterpolation) {
-      var last = (this as StringInterpolation).elements.last;
-      if (last is InterpolationExpression) {
-        // Treat an interpolation expression as containing whitespace so as to
-        // avoid over-reporting strings that end with an interpolation
-        // expression.
-        return true;
-      } else if (last is InterpolationString) {
-        return last.value.endsWithWhitespace;
-      }
+    var self = this;
+    if (self is SimpleStringLiteral) {
+      return self.value.endsWithWhitespace;
+    } else if (self is StringInterpolation) {
+      var last = self.elements.last as InterpolationString;
```

Comment:
Not necessary, but you could simplify this further by using `self.lastString` which casts the element for you.